rct229/rules/section23/section23rule6.py

```diff
@@ -0,0 +1,105 @@
+    baseline_system_type_compare,
+)
+from rct229.ruleset_functions.baseline_systems.baseline_system_util import HVAC_SYS
+from rct229.ruleset_functions.get_baseline_system_types import get_baseline_system_types
+from rct229.utils.pint_utils import CalcQ
+from rct229.utils.std_comparisons import std_equal
+
+APPLICABLE_SYS_TYPES = [
+    HVAC_SYS.SYS_6,
+    HVAC_SYS.SYS_8,
+]
+
+
+class Section23Rule6(RuleDefinitionListIndexedBase):
+    """Rule 6 of ASHRAE 90.1-2019 Appendix G Section 23 (Air-side)"""
+
+    def __init__(self):
+        super(Section23Rule6, self).__init__(
+            rmrs_used=UserBaselineProposedVals(False, True, False),
+            each_rule=Section23Rule6.TerminalRule(),
+            index_rmr="baseline",
+            id="23-6",
+            description="For baseline systems 6 and 8, Fans in parallel VAV fan-powered boxes shall be sized for 50% of the peak design primary air (from the VAV air-handling unit) flow rate and shall be modeled with 0.35 W/cfm fan power.",
+            ruleset_section_title="HVAC - Airside",
+            standard_section="Section G3.1.3.14 Fan Power and Control (Systems 6 and 8)",
+            is_primary_rule=True,
+            rmr_context="ruleset_model_instances/0",
+            list_path="$..terminals[*]",
+        )
+
+    def is_applicable(self, context, data=None):
+        rmi_b = context.baseline
+        # baseline_system_types_dict = get_baseline_system_types(rmi_b)
+        baseline_system_types_dict = {
+            "Sys-6": ["System 6"],
+            "Sys-8": [],
+        }  # This will be removed once PR #899 is merged
+
+        return any(
+            [
+                baseline_system_type_compare(system_type, applicable_sys_type, False)
+                for system_type in baseline_system_types_dict.keys()
+                for applicable_sys_type in APPLICABLE_SYS_TYPES
+            ]
+        )
+
+    def create_data(self, context, data):
+        rmi_b = context.baseline
+        # baseline_system_types_dict = get_baseline_system_types(rmi_b)
+        baseline_system_types_dict = {
+            "Sys-6": ["System 6"],
+            "Sys-8": [],
+        }  # This will be removed once PR #899 is merged
+        applicable_hvac_sys_ids = [
+            hvac_id
+            for sys_type in APPLICABLE_SYS_TYPES
+            for hvac_id in baseline_system_types_dict[sys_type]
+        ]
+
+        return {"applicable_hvac_sys_ids": applicable_hvac_sys_ids}
+
+    def list_filter(self, context_item, data):
+        applicable_hvac_sys_ids = data["applicable_hvac_sys_ids"]
+
+        return context_item.baseline["id"] in applicable_hvac_sys_ids
+
+    class TerminalRule(RuleDefinitionBase):
+        def __init__(self):
+            super(Section23Rule6.TerminalRule, self).__init__(
+                rmrs_used=UserBaselineProposedVals(False, True, False),
+                required_fields={
+                    "$": ["primary_airflow"],
```

Comment:
Do we also need to require the "fan" field?

Reply:
Great catch! Fixed.